docs/literate/src/files/hohqmesh_literate.jl

```diff
@@ -0,0 +1,521 @@
+# The name for the `CHAIN` is used internally by HOHQMesh, so if you have multiple `CHAIN`s they **must be given a unique name**.
+# The name for the `PARAMETRIC_EQUATION_CURVE` will be printed to the appropriate boundaries within the `.mesh` file produced by
+# HOHQMesh. The resulting mesh generated by HOHQMesh.jl is given in the following figure.
+
+# ![mesh_straight_sides](https://user-images.githubusercontent.com/25242486/129603958-08e4b874-53d5-4511-9a54-6daf4c21edca.png)
+
+# The mesh file `ice_cream_straight_sides.mesh` and its associated file for plotting
+# `ice_cream_straight_sides.tec` are placed in the `out` folder.
+# We note that Trixi.jl uses the boundary name information from the control file
+# to assign boundary conditions in an elixir file.
+# Therefore, the name should start with a letter and consist only of alphanumeric characters and underscores. Please note that the name will be treated as case sensitive.
+
+# ## Example simulation on `ice_cream_straight_sides.mesh`
+
+# With this newly generated mesh we are ready to run a Trixi.jl simulation on an unstructured quadrilateral mesh.
+# For this we must create a new elixir file.
+
+# The elixir file given below creates an initial condition for a
+# uniform background flow state with a free stream Mach number of 0.3.
+# A focus for this part of the tutorial is to specify the boundary conditions and to construct the new mesh from the
+# file that was generated in the previous exercise.
+
+# It is straightforward to set the different boundary
+# condition types in an elixir by assigning a particular function to a boundary name inside a
+# Julia dictionary, `Dict`, variable. Observe that the names of these boundaries match those provided by HOHQMesh
+# either by default, e.g. `Bottom`, or user assigned, e.g. `IceCream`. For this problem setup use
+# * Freestream boundary conditions on the four box edges.
+# * Free slip wall boundary condition on the interior curved boundaries.
+
+# We create the mesh file `ice_cream_straight_sides.mesh` and its associated file for plotting
+# `ice_cream_straight_sides.tec` by using HOHQMesh's function `generate_mesh`.
+using HOHQMesh
+control_file = joinpath(@__DIR__, "..", "ice_cream_straight_sides.control")
+control_file = joinpath("docs", "src", "ice_cream_straight_sides.control") #src
+output = generate_mesh(control_file);
+
+# To construct the unstructured quadrilateral mesh from the HOHQMesh file we point to the appropriate location
+# with the variable `mesh_file` and then feed this into the constructor for the `UnstructuredMesh2D` type in Trixi.jl
+
+## create the unstructured from your mesh file
+using Trixi
+mesh_file = joinpath(@__DIR__, "out", "ice_cream_straight_sides.mesh") #nb
+mesh_file = joinpath(@__DIR__, "..", "..", "..", "out", "ice_cream_straight_sides.mesh") #md
+mesh_file = joinpath("out", "ice_cream_straight_sides.mesh") #src
+mesh = UnstructuredMesh2D(mesh_file);
```

Comment:
It's a bit ugly to get the right link in the right file right now. That's because the code is executed 3 times. First, when execute the file for testing. Second, when Documenter executes the `@example` blocks and third, when one manually calls it in the notebook.
If there is any other and less hacky solution, that would be great.

Reply:
Shouldn't the `"out"` folder be created in the current working directory? If so, shouldn't `joinpath("out", "ice_cream_straight_sides.mesh")` work in all cases? Or do I miss something?

Reply:
Yes. I was confused because of the two working directories and the location of the `.control` file. Now everything is collected in the `out` folder and created with only one line each.